### ISSUE TITLE
Caching with sidekick is currently broken

### ISSUE THREAD

**Issue by vrajanap** [minio/sidekick#57]
While trying out sidekick caching behavior, I noticed sidekick's requests to minio server fails due to `unexpected EOF` and with that sidekick is unable to cache the objects

A short repro: 
https://gist.github.com/vrajanap/d261a90ba159a67047ed480325913889

## Expected Behavior
Sidekick successfully caches the object and then is able to reuse the cached object. 

## Current Behavior
Though the caching server is able to write the object to disk, it fails with `unexpected EOF` and then sidekick is unable to use the object. 

## Steps to Reproduce (for bugs)
1. Download the docker-compose.yml file from https://gist.github.com/vrajanap/d261a90ba159a67047ed480325913889. 
2. docker-compose -f path-to-downloaded-docker_compose.yml up
3. See the exception coming in from `cache` container as in https://gist.githubusercontent.com/vrajanap/d261a90ba159a67047ed480325913889/raw/cb4cc77ea7b7b60c42bf8ac129fcdce65dae022f/output.txt. 

## Context
Trying to setup cache server for sidekick 

## Regression
Unclear if this is regression. Tried with an old minio server from Oct 2020 and was still able to repro the issue. 

## Your Environment

* Version used (`sidekick version`): 1.0.0
* Link to your project: https://gist.github.com/vrajanap/d261a90ba159a67047ed480325913889
```sh
$ docker run quay.io/minio/minio --version
minio version RELEASE.2021-09-09T21-37-07Z
$ docker run minio/sidekick --version
/sidekick version 1.0.0
```

**Comment by vrajanap** [comment]
Hi @harshavardhana ! Are you looking into this? Please let me know if you need additional data with the repro.

**Comment by harshavardhana** [comment]
> Hi @harshavardhana ! Are you looking into this? Please let me know if you need additional data with the repro.

There are no plans on my end to look into this. This is not a priority at the moment.

**Comment by harshavardhana** [comment]
Fixed in latest release v1.1.0
